__tests__/end-to-end.js

```diff
@@ -510,6 +557,16 @@ async function click (selector: string) {
   await page.click(selector) // , {delay: 3})
 }
 
+async function elementClick (elementHandle: any, selector: string) {
+  log.info(`finding selector: ${selector} in element handle ${elementHandle}`)
+  const selectedElement = await elementHandle.$(selector)
+  if (!selectedElement) {
+    throw new Error(`Could't find "${selector}" within elementHandle ${elementHandle}`)
+  }
+  log.info(`clicking selector: ${selector} of element handle: ${elementHandle}`)
+  await selectedElement.click() // , {delay: 3})
+}
+
```

Comment:
I'm getting a little confused with all of these helper functions. What is the difference between, say, `elementClick` and `waitForAndClick`? When should each be used? Might be good to add some jsdoc.